modules/ethereum/src/verification.rs

```diff
@@ -362,41 +367,88 @@ mod tests {
 	}
 
 	fn verify_with_config(config: &AuraConfiguration, header: &Header) -> Result<ImportContext<AccountId>, Error> {
-		let storage = InMemoryStorage::new(genesis(), validators_addresses(3));
-		verify_aura_header(&storage, &config, None, header)
+		custom_test_ext(genesis(), validators_addresses(3)).execute_with(|| {
+			let storage = BridgeStorage::<TestRuntime>::new();
+			verify_aura_header(&storage, &config, None, header)
+		})
 	}
 
 	fn default_verify(header: &Header) -> Result<ImportContext<AccountId>, Error> {
-		verify_with_config(&kovan_aura_config(), header)
+		verify_with_config(&test_aura_config(), header)
 	}
 
 	fn default_accept_into_pool(
-		mut make_header: impl FnMut(&mut InMemoryStorage, &[KeyPair]) -> (Header, Option<Vec<Receipt>>),
+		mut make_header: impl FnMut(&[KeyPair]) -> (Header, Option<Vec<Receipt>>),
 	) -> Result<(Vec<Vec<u8>>, Vec<Vec<u8>>), Error> {
-		let validators = vec![validator(0), validator(1), validator(2)];
-		let mut storage = InMemoryStorage::new(genesis(), validators_addresses(3));
-		let block1 = block_i(&storage, 1, &validators);
-		storage.insert(block1);
-		let block2 = block_i(&storage, 2, &validators);
-		let block2_hash = block2.hash();
-		storage.insert(block2);
-		let block3 = block_i(&storage, 3, &validators);
-		let block3_hash = block3.hash();
-		storage.insert(block3);
-		storage.set_finalized_block((2, block2_hash));
-		storage.set_best_block((3, block3_hash));
-
-		let validators_config =
-			ValidatorsConfiguration::Single(ValidatorsSource::Contract(Default::default(), Vec::new()));
-		let (header, receipts) = make_header(&mut storage, &validators);
-		accept_aura_header_into_pool(
-			&storage,
-			&kovan_aura_config(),
-			&validators_config,
-			&pool_configuration(),
-			&header,
-			receipts.as_ref(),
-		)
+		custom_test_ext(genesis(), validators_addresses(3)).execute_with(|| {
+			let validators = vec![validator(0), validator(1), validator(2)];
+			let mut storage = BridgeStorage::<TestRuntime>::new();
+			let block1 = block_i(1, &validators);
+			storage.insert_header(
+				storage
+					.import_context(None, &block1.parent_hash)
+					.unwrap()
+					.into_import_header(true, block1.hash(), block1, 0.into(), None, None),
```

Comment:
I think this can be done in a function. It would make the test flow in a similar way as before